server/views/account/signin.jade

```diff
@@ -23,3 +23,33 @@ block content
         p
             a(href="/email-signin") If you originally signed up using your email address, you can sign in here.
 
+    script.
+      $(document).ready(function() {
+        var method = localStorage.getItem('lastSigninMethod'),
+          btnSelector = 'a.btn.btn-lg.btn-block.btn-social';
+        if (method) {
+          try {
+            var obj = JSON.parse(method);
+            $.each($(btnSelector), function(i, item) {
```

Comment:
We just need to wrap the parse method since you can store anything in localStorage.
```js
try {
  var obj = JSON.parse(method);
} catch () {
  console.error('Invalid sign in thingy stored', method); // probably a better error message should go here.
  return; // exit function early
}
```